### DISCUSSION TITLE
Keep download and other buttons when loading preview

### DISCUSSION THREAD

**Discussion by myl7** [spencerwooo/onedrive-vercel-index#482] [Ideas]
When loading preview, `<DownloadBtnContainer>` is not rendered though the app has got all required information of it. I would like to suggest add `<DownloadBtnContainer>` even when flags that indicate loading state like `validating` or `loading` is true.

**Comment by spencerwooo** [comment]
Valid, good idea 👍

**Comment by spencerwooo** [comment]
Seems only text, markdown, and code previews are effected? Or am I missing something? (9e50a024eedb25308548897e64a3ca880ad03ca2)

**Reply by myl7** [reply]
https://github.com/spencerwooo/onedrive-vercel-index/blob/9e50a024eedb25308548897e64a3ca880ad03ca2/components/previews/TextPreview.tsx#L23-L29

https://github.com/spencerwooo/onedrive-vercel-index/blob/9e50a024eedb25308548897e64a3ca880ad03ca2/components/previews/URLPreview.tsx#L30-L36

How about these positions?

**Reply by spencerwooo** [reply]
first one yes, second one no. url preview button is a link and requires the file content to be downloaded to render.

**Reply by myl7** [reply]
Users may want to download the url file ... Well a url file is useless. You are right.
